src/nvim/testdir/test_autocmd.vim

```diff
@@ -0,0 +1,8 @@
+" Tests for autocommands
+
+func Test_vim_did_enter()
+  call assert_false(v:vim_did_enter)
+
+  " This script will never reach the main loop, can't check if v:vim_did_enter
+  " becomes one.
```

Comment:
It doesn't look like we can test it either.  I tried adding a lua test and it fails:

```diff
@@ -28,4 +28,14 @@ describe('autocmds:', function()
     command('tabnew')
     assert.same(expected, eval('g:foo'))
   end)
+
+  describe('VimEnter', function()
+    before_each(function()
+      clear({'--cmd', 'autocmd VimEnter * let g:did_enter = v:vim_did_enter'})
+    end)
+
+    it('sees v:vim_did_enter as 1', function()
+      eq(1, eval('g:did_enter'))
+    end)
+  end)
 end)
```

```
[0/1] cd /home/mccoyj1/src/neovim/build && /usr/bin/cmake -DBUSTED_PRG=/u...onal -DSYSTEM_NAME=Linux -P /home/mccoyj1/src/neovim/cmake/RunTests.cmak
# test/functional/autocmd/autocmd_spec.lua @ 10: autocmds: :tabnew triggers events in the correct order
ok 1 - autocmds: :tabnew triggers events in the correct order
# test/functional/autocmd/autocmd_spec.lua @ 37: autocmds: VimEnter sees v:vim_did_enter as 1
not ok 2 - autocmds: VimEnter sees v:vim_did_enter as 1
# test/functional/autocmd/autocmd_spec.lua @ 37
# Failure message: ./test/functional/helpers.lua:84: Failed to evaluate expression
# stack traceback:
#       ./test/functional/helpers.lua:84: in function 'eval'
#       test/functional/autocmd/autocmd_spec.lua:38: in function <test/functional/autocmd/autocmd_spec.lua:37>
# 
1..2
```